google/cloud/bigtable/internal/async_bulk_apply_test.cc

```diff
@@ -288,6 +285,179 @@ TEST_F(AsyncBulkApplyTest, AsyncBulkApplyFailureRetry) {
+        return reader;
+      })
+      .WillOnce([](grpc::ClientContext*, btproto::MutateRowsRequest const&,
+                   grpc::CompletionQueue*) {
+        auto reader = absl::make_unique<
+            MockClientAsyncReaderInterface<btproto::MutateRowsResponse>>();
+        EXPECT_CALL(*reader, Read)
+            .WillOnce([](btproto::MutateRowsResponse* r, void*) {
+              auto& r1 = *r->add_entries();
+              r1.set_index(0);
+              r1.mutable_status()->set_code(grpc::StatusCode::OK);
+
+              auto& r2 = *r->add_entries();
+              r2.set_index(1);
+              r2.mutable_status()->set_code(grpc::StatusCode::OK);
+            })
+            .WillOnce([](btproto::MutateRowsResponse*, void*) {});
+        EXPECT_CALL(*reader, Finish).WillOnce([](grpc::Status* status, void*) {
+          *status = grpc::Status::OK;
+        });
+        EXPECT_CALL(*reader, StartCall);
+        return reader;
+      });
+
+  auto bulk_apply_future = internal::AsyncRetryBulkApply::Create(
+      cq_, rpc_retry_policy_->clone(), std::move(mock),
+      *idempotent_mutation_policy_, metadata_update_policy_, client_,
+      "my-app-profile", "my-table", std::move(mut));
+
+  SimulateIterationThatRetries();
+  ASSERT_EQ(1U, cq_impl_->size());
+
+  SimulateIteration();
+  ASSERT_EQ(0U, cq_impl_->size());
+}
+
+TEST_F(AsyncBulkApplyTest, CancelDuringBackoff) {
+  bigtable::BulkMutation mut{
+      bigtable::SingleRowMutation("foo2",
+                                  {bigtable::SetCell("f", "c", 0_ms, "v2")}),
+      bigtable::SingleRowMutation("foo3",
+                                  {bigtable::SetCell("f", "c", 0_ms, "v3")}),
+  };
+
+  auto grpc_error = grpc::Status(grpc::StatusCode::UNAVAILABLE, "try again");
+  auto error = MakeStatusFromRpcError(grpc_error);
+
+  std::unique_ptr<MockBackoffPolicy> mock(new MockBackoffPolicy);
+  EXPECT_CALL(*mock, Setup);
+  EXPECT_CALL(*mock, OnCompletion(error)).WillOnce([](Status const&) {
+    return 10_ms;
+  });
+
+  EXPECT_CALL(*client_, PrepareAsyncMutateRows)
+      .WillOnce([grpc_error](grpc::ClientContext*,
+                             btproto::MutateRowsRequest const&,
+                             grpc::CompletionQueue*) {
+        auto reader = absl::make_unique<
+            MockClientAsyncReaderInterface<btproto::MutateRowsResponse>>();
+        EXPECT_CALL(*reader, Read).Times(2);
+        EXPECT_CALL(*reader, Finish)
+            .WillOnce([grpc_error](grpc::Status* status, void*) {
+              *status = grpc_error;
+            });
+        EXPECT_CALL(*reader, StartCall);
+        return reader;
+      });
+
+  auto bulk_apply_future = internal::AsyncRetryBulkApply::Create(
+      cq_, rpc_retry_policy_->clone(), std::move(mock),
+      *idempotent_mutation_policy_, metadata_update_policy_, client_,
+      "my-app-profile", "my-table", std::move(mut));
+
+  SimulateIteration();
+  ASSERT_EQ(1U, cq_impl_->size());
+
+  // cancel the future
```

Comment:
I would say "cancel the pending operation"

Reply:
Done.